zrml/swaps/src/tests.rs

```diff
@@ -1883,6 +1882,26 @@ fn create_pool_transfers_the_correct_amount_of_tokens() {
     });
 }
 
+#[test]
+fn close_pool_fails_if_pool_does_not_exist() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_noop!(Swaps::close_pool(0), crate::Error::<Runtime>::PoolDoesNotExist);
+    });
+}
+
+#[test]
+fn close_pool_succeeds_and_emits_correct_event_if_pool_exists() {
+    ExtBuilder::default().build().execute_with(|| {
+        frame_system::Pallet::<Runtime>::set_block_number(1);
+        create_initial_pool(ScoringRule::CPMM, true);
+        let pool_id = 0;
+        assert_ok!(Swaps::close_pool(pool_id));
+        let pool = Swaps::pool(pool_id).unwrap();
+        assert_eq!(pool.pool_status, PoolStatus::Closed);
+        System::assert_last_event(Event::PoolClosed(pool_id).into());
+    });
+}
+
```

Comment:
A test that verifies that a pool does only valid state transitions would be nice, i.e. only `Active -> Closed` is allowed.
I am wondering if we should signal that a pool is cleaned up with one addition state. Would make sense to know which pools were cleaned up and which were not (might be useful when we write migrations in the future or other scenarios).

Reply:
Implemented in d553b30e3d9bd7c5f3e272aed281c4c1ed1021b3 and 3f3bbad6a2b82040e28d5aaa33d91d42a8e8121d.

(I noticed too late that I could have just renamed `Stale` to `Clean`.) I tested the migration against Battery Station. Passed.